### ISSUE TITLE
Ability to revert drag/sort if desired from within sortupdate callback

### ISSUE THREAD

**Issue by coreypmurphy** [lukasoppermann/html5sortable#90]
Intercepting "sortupdate" is a great feature if you use it to update a dataset/datastore or do some other permanent/persistent action.  However if you want conditionalize the "finality" of the element placement you can't.  It would be nice to be able to revert the drag and sort from within the "sortupdate" callback if desired.

In my case I want to be able to open a dialog and accept data entry that is contextually specific to the drag/drop/sort operation.  If the user cancels that data entry I want to revert the dragged element, and if they submit the data, persist the elements new location and it's accompanying data.

I can accomplish this "intercept and revert" using jQueryUI's "drop" event handler but only on "draggable/droppable" elements and not the combined "draggable/droppable/sortable" that this plugin provides as I'd like to be able to drag and drop and sort between multiple lists.

**Comment by lukasoppermann** [comment]
Could you not store the original position of the element and the original parent and on a canceled event, just move it back?

I think remembering the original position and parent until the page is refreshed is something we could definitely consider, however at the moment I feel the whole reverting is a little "out of scope" for the plugin.

If you have a solution which does not introduce to much complexity I will gladly see if it can be merged. Should you decide to send a PR, please write a separate function which can be called, as I will be moving towards refactoring the whole plugin to be split into functions for easy testing, extending and refactoring.

**Comment by lukasoppermann** [comment]
@coreypmurphy Actually this should currently be possible this is the current data for sortupdate:

``` javascript
item: dragging,
oldindex: index,
startparent: startParent,
endparent: newParent
```

You could just bind an event to sortupdate and pull up a dialog, of the user cancels you could use `oldindex` and `startparent` to revert the item. Would that work for you or am I missing something here?

**Comment by coreypmurphy** [comment]
Lukas,

Thanks for all your hard work recently. You appear to be really getting after it with this project and it's cool to see that. I'll pull down the latest and see if I can accomplish what you suggested and let you know. 

Thanks again,

Corey P. Murphy

**Comment by lukasoppermann** [comment]
@coreypmurphy perfect, thanks. If this can not be accomplished, I would welcome a PR, but please make sure to implement it in a way that will have the least performance influence possible. :)

**Comment by lukasoppermann** [comment]
Hey @coreypmurphy any news on this? Just letting you know, that the api for `sortupdate` will change a bit once this PR is merged: https://github.com/voidberg/html5sortable/pull/139

**Comment by coreypmurphy** [comment]
@lukasoppermann I've pulled down the latest but I'm still having an issue.  Probably just fried at the end of the day at work but I'm not able to revert it still.  I've set the "index" of the item in question to the "oldindex" and then tried to reload the sortable but it's not reloading it.  The example below (minus the markup) assumes a pair of connected lists however I'm only reordering within one of those lists so I'm only setting the index of the dragged element to the oldindex position and then attempting to reload both lists.

``` javascript
var $item;
var $itemDragged;
var origin;
var destination;

$('.connected').sortable({
    connectWith: '.connected'
  }).bind('sortupdate', function(e, ui) {    
    $item = ui;
    $itemDragged = ui.item;
    origin = ui.startparent[0].id;
    destination = this.id;
    $("#form_to_complete").show('fade');
  });

$("#save").on("click", function() {
    doSomeDBStuff(origin,destination,$itemDragged);
    $("#form_to_complete").hide();
});

$(".close").on("click", function(){    
    $item.item.index = $item.oldindex;
    $('.connected').sortable();
    $("#form_to_complete").hide();
});
```

Thanks again for all the work and help.  Thoughts?

**Comment by coreypmurphy** [comment]
Forgot to mention that in the code above I was assuming that start and end parent are the same for testing purposes.

**Comment by lukasoppermann** [comment]
Hey @coreypmurphy I did not have time to check it out yet. I will hopefully get to it this weekend.

**Comment by lukasoppermann** [comment]
Hey @coreypmurphy, the sortable is not as advanced as you think. :) You can not just set the index of an item and reload. You would need to do some jQuery stuff:

``` javascript
$('.connected').sortable({
    connectWith: '.connected'
  }).bind('sortupdate', function(e, ui) {    
    $item = ui;
    $itemDragged = ui.item;
    origin = ui.startparent;
    $("#form_to_complete").show('fade');
  });
$(".close").on("click", function(){
    // this code is untested, but you will get the idea. 
    origin.children(origin.data('items')).get($item.oldIndex - 1).after($itemDragged);
    $("#form_to_complete").hide();
});
```

**Comment by coreypmurphy** [comment]
Ah.  Sorry.  I guess I figured it behaved similar to the jQuery "revert" but I guess in this case the element has already beed dropped and the list sorted so I'll have to perform the "revert" manually.  Cool.  I greatly appreciate your time.  Awesome to see you taking ownership of this project.

**Comment by lukasoppermann** [comment]
No worries, if you could share a version of your revert function via jsfiddle, I would greatly appreciate it.

**Comment by coreypmurphy** [comment]
Sure thing. Would be good to see it incorporated as a method into the plugin. I'd gladly send a pull request but I'm extremely swamped. 

Corey P. Murphy

**Comment by lukasoppermann** [comment]
Sure that would be awesome.
